system/security/shared/caching.cpp

```diff
@@ -585,6 +562,46 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
         return true;
     }
 
+    // If the cache is not ready, then an initial fill is required. Acquire the cache RW lock and make sure no other thread has
+    // tried to fill the cache. Once filled, start the fill thread and wait for it to be ready.
+    if (!m_fileScopeCacheReady)
+    {
+        WriteLockBlock writeLock(m_scopesRWLock);
+
+        if (!m_fileScopeCacheReady)
+        {
+            IArrayOf<ISecResource> scopes;
+            aindex_t count = m_secMgr->getManagedScopeTree(RT_FILE_SCOPE, nullptr, scopes);
+            removeAllManagedFileScopes();
+            if (count)
+                addManagedFileScopes(scopes);
+            time_t now;
+            time(&now);
+            m_lastCacheFillTime = now;
+
+            m_fileScopeCacheFillThread = std::thread(&CPermissionsCache::managedFileScopesCacheFillThread, this);
```

Comment:
Should not happen if timeoutSeconds is 0.  Do we want to support that option at all??

Reply:
>Do we want to support that option at all??

No, the code would be simpler and safer if it didn't (already somewhat assumes it's non-zero in some places). See other comments.